streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/python/PythonGateway.java

```diff
@@ -0,0 +1,143 @@
+package org.ray.streaming.runtime.python;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.msgpack.core.Preconditions;
+import org.ray.api.annotation.RayRemote;
+import org.ray.streaming.api.context.StreamingContext;
+import org.ray.streaming.python.PythonFunction;
+import org.ray.streaming.python.PythonPartition;
+import org.ray.streaming.python.stream.PythonStreamSource;
+import org.ray.streaming.util.ReflectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Gateway for streaming python api.
+ * All calls on DataStream in python will be mapped to DataStream call in java by this
+ * PythonGateway using ray calls.
+ * <p>
+ * Note: this class needs to be in sync with `ray.streaming.gateway_client.GatewayClient`
+ */
+@SuppressWarnings("unchecked")
+@RayRemote
+public class PythonGateway {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PythonGateway.class);
+  private static final String REFERENCE_ID_PREFIX = "__gateway_reference_id__";
+
+  private MsgPackSerializer serializer;
+  private Map<String, Object> referenceMap;
+  private StreamingContext streamingContext;
+
+  public PythonGateway() {
+    serializer = new MsgPackSerializer();
+    referenceMap = new HashMap<>();
+    LOGGER.info("PythonGateway created");
+  }
+
+  public byte[] createStreamingContext() {
+    streamingContext = StreamingContext.buildContext();
+    LOGGER.info("StreamingContext created");
+    referenceMap.put(getReferenceId(streamingContext), streamingContext);
+    return serializer.serialize(getReferenceId(streamingContext));
+  }
+
+  public byte[] withConfig(byte[] confBytes) {
```

Comment:
```suggestion
  public byte[] withConfig(byte[] config) {
```

Reply:
```
  public byte[] withConfig(byte[] confBytes) {
    Preconditions.checkNotNull(streamingContext);
    try {
      Map<String, String> config = (Map<String, String>) serializer.deserialize(confBytes);
      LOG.info("Set config {}", config);
      streamingContext.withConfig(config);
      return new byte[1];
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }
```
Using `confBytes` can distinguish with `config` in method body.

Reply:
It is about the naming convention. If you want to use the pattern <name>Bytes, then all methods with the byte[] input parameter should follow that pattern. Since there are a lot of such methods and it turns out the "Bytes" is kinds of redundant, because developer can get that information from the type of the input parameter, i.e. "byte[]".